### ISSUE TITLE
Mise en place d'une VM pour le groupe OSM Plein Air

### ISSUE THREAD

**Issue by pyrog** [osm-fr/infrastructure#464]
Le groupe a besoin d'une VM sous **Debian** pour mettre en place temporairement une instance de [waymarked trails](https://hiking.waymarkedtrails.org/#?map=8.0/48.7949/2.491).
Elle serait déployée sous le nom de domaine [pleinair.openstreetmap.fr](https://pleinair.openstreetmap.fr) (à défaut [outdoor.openstreetmap.fr](https://outdoor.openstreetmap.fr)).

Il faudrait à minima un accès SSH pour [moi](https://www.openstreetmap.org/user/pyrog) et [Stéphane](https://www.openstreetmap.org/user/StC) (@StC-OSM).

Le but est d'améliorer le code, le rendu, tester de nouvelles fonctionnalités.

Nous avons besoin de 150 Go  pour charger les données de la France.
Moins si on se limite à une région.

Merci d'avance 😃 

PS: Voici les caractéristiques indiquées par [Sarah Hoffmann](https://github.com/lonvia).

> La base de données pour la planète fait 310 GO + 80 GO pour les positions de
> noeuds. Pour la France ça sera beaucoup moins, peut-être 30 + 80 GO.
> 
> Je fais très peu de 'caching' de tuiles. 7 GO pour la planète en ce moment. La plupart des tuiles sont créées sur demande.
> 
> Si tu veux avoir le DEM, tu aura besoin de 50 GO de plus.
>

**Comment by jocelynj** [comment]
@pyrog : on partirait sur un container dédié pour la base postgresql - est-ce que ça marcherait pour l'instance de waymarked trails ?

**Comment by pyrog** [comment]
En théorie oui.
Sarah m'a montré le début de l'installation de WMT (backend) en partant d'une VM Debian puis en rajoutant Postgresql, PostGIS…

Mais en pratique j'ai peur qu'on soit bloqué par pleins de détails et qu'au final on n'arrive pas à faire l'installation de WMT.

Est-ce que tu peux installer un Debian "vierge" dans un premier temps ?
On (Stéphane et moi) reproduit la procédure d'installation de Sarah, et une fois qu'on a compris comment tout ça fonctionne on peut retenter sur un conteneur pour Postgresql ?

Je peux aussi lui demander son avis. As-tu une description de ce conteneur ?

**Comment by jocelynj** [comment]
Pas de souci: on va démarrer avec 2 container:
  - pleinair.vm.openstreetmap.fr, où tu pourras tester l'install complète, avec une petite région de préférence
  - postgres-ovh.vm.openstreetmap.fr, où on mettra la base après réinstall

La première est en cours d'install, mais je n'ai pas créé de compte pour Stéphane, vu que je n'ai pas sa clé ssh.

**Comment by StC-OSM** [comment]
Hello !

La voici : 

ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHAtO6RudZerjZ2EgbqlN+395fJQPCkZBMSgBKT4QAYD StC-OSM

St.

**Comment by jocelynj** [comment]
J'ai ajouté ton compte, avec accès sudo. Pour info, la VM n'est accessible qu'en ipv6 - merci de me dire si il faut un pont ipv4-ipv6 pour l'accès.

Pour le site web pleinair.openstreetmap.fr, vous avez besoin que je créé l'entrée avant que ce ne soit installé ?

**Comment by Marc-marc-marc** [comment]
je laisse @jocelynj faire la partie dns vu que je n'ai pas l'ip

tu as besoin de rebond  ipv4 vers ipv6 uniquement si ton fournisseur est à la préhistoire (taquinerie vu qu'on est en 2023) et n'as pas d'ipv6 fonctionnelle, sinon tu n'en as pas besoin

ce sont des CT vu par debian comme une machine, même si c'est tout a fait détectable que c'est une machine lxc et non physique. cependant cela n'a en pratique pas d'importance.
docker est activable si @jocelyn ne l'a pas fait

**Comment by pyrog** [comment]
On a réglé nos mac en IP V6. Stéphane sa box et moi le protocole APN de mon téléphone.
Ça passe 😄 
(Il faut juste garder APN V4/V6 car GitHub entre-autres et en IP V4 uniquement)

**Comment by pyrog** [comment]
Je ne retrouve pas ma [clé privée utilisée en décembre 2020 pour Mobilizon](https://github.com/osm-fr/infrastructure/issues/253#issuecomment-748493374), ni le compte utilisateur.

Pouvez-vous utiliser celle-ci ?
Et pour le compte soit yves, soit pyrog (qui est mon pseudo OSM) ?

**Comment by jocelynj** [comment]
Le compte est pyrog, et je viens de t'ajouter cette clé ssh.

**Comment by StC-OSM** [comment]
Bonjour,

et merci pour la mise à disposition du serveur virtuel pleinair.vm.openstreetmap.fr <http://pleinair.vm.openstreetmap.fr/> ! Nous venons de réussir à faire tourner une instance de développement de Waymarkedtrails, ce qui devrait nous permettre d’expérimenter des évolutions du code. 

Maintenant que l’appli tourne, on peut envisager de finaliser la configuration réseau. 

1. DNS

Actuellement, une configuration de nginx permet d’accéder à l’application sur http://pleinair.vm.openstreetmap.fr/waymarkedtrails/.

Idéalement, on permettrait l’accès à l’application depuis https://pleinair.openstreetmap.fr/waymarkedtrails. J’imagine qu’il y a une manip de DNS et de certificat à faire ?


2. ports et sécurité

Un coup d’œil rapide avec ufw semble indiquer que de nombreux ports sont accessibles sur la machine. Faut-il sécuriser le serveur en fermant des ports ? Ou bien y a-t-il un firewall ailleurs que sur la machine elle-même ?

A bientôt,

St.

**Comment by tykayn** [comment]
je crois qu'il y a un système de bastion pour sécuriser les accès, mais il faudrait que quelqu'un d'autre confirme.

concernant ce ticket je pense qu'il a été résolu maintenant que le serveur est fonctionnel, on peut donc fermer.

**Comment by jocelynj** [comment]
Je vois qu’on n’a pas fini la configuration de ce serveur, désolé :( 

Est-ce qu’il manque juste la config du DNS ?
Et est-ce que ça vous embête de changer de machine avant d’activer ?

**Comment by StC-OSM** [comment]
Bonjour,

si si, ce serveur a fonctionné et donné satisfaction pendant plusieurs mois. Et lorsqu’il a cessé de fonctionner, nous aurions dû alerter mais comme souvent avec OSM l’appel des ajouts dans la base au quotidien a été plus fort :-)

Aucun pb pour le changer de machine, mais allons-nous retrouver nos données en l’état ?

A bientôt,

St.

**Comment by jocelynj** [comment]
Oui, ça sera une migration en gardant les données: la seule chose qui devrait changer, c’est l’ip pour accéder à la machine, si tout se passe bien :)

Je rajouterai une entrée pleinair.openstreetmap.fr après migration.

**Comment by jocelynj** [comment]
La VM a été migré sur un autre cluster, et http://pleinair.openstreetmap.fr configuré sur le proxy.

Je vous laisse fermer le ticket quand le site marche correctement ?

**Comment by StC-OSM** [comment]
Merci beaucoup ! Il nous reste à tester et réactiver les actions en cours de notre côté.

A bientôt,

St.

**Comment by jocelynj** [comment]
C’est un souci côté VM, pas côté proxy. À ce que je vois, nginx ne démarre pas à cause d’une config erronée:
```
Jul 28 19:40:29 pleinair systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Jul 28 19:40:29 pleinair nginx[2919295]: nginx: [emerg] bind() to 0.0.0.0:35729 failed (98: Address already in use)
```

Mais vu qu’on n’a pas de nouvelles de @StC-OSM , je ne sais pas si pleinair est vraiment maintenu.

**Comment by cquest** [comment]
Je ferme cette issue qui concernait la mise en place du conteneur

Suite sur https://github.com/osm-fr/infrastructure/issues/637
